src/mcts/search.cc

```diff
@@ -868,47 +911,59 @@ void SearchWorker::FetchMinibatchResults() {
 void SearchWorker::DoBackupUpdate() {
   // Update nodes.
   SharedMutex::Lock lock(search_->nodes_mutex_);
-  for (NodeToProcess& node_to_process : nodes_to_process_) {
-    Node* node = node_to_process.node;
-    if (node_to_process.is_collision) {
-      // If it was a collision, just undo counters.
-      for (node = node->GetParent(); node != search_->root_node_->GetParent();
-           node = node->GetParent()) {
-        node->CancelScoreUpdate();
-      }
-      continue;
+
+  // If it's out of order eval, only backpropagate last item in a batch.
+  if (out_of_order_eval_) {
+    DoBackupUpdateSingleNode(nodes_to_process_.back());
+    return;
+  }
+
+  for (const NodeToProcess& node_to_process : nodes_to_process_) {
+    DoBackupUpdateSingleNode(node_to_process);
+  }
+}
+
+void SearchWorker::DoBackupUpdateSingleNode(
+    const NodeToProcess& node_to_process) REQUIRES(search_->nodes_mutex_) {
```

Comment:
Breaking the line here seems awkward to me, why not 

```
void Class::Function(arg)
                  REQUIRES(foo) {
```
?

(Again, ancillary. If clang did this automatically, then I guess ignore this comment.)

Reply:
That's clang. I usually write everything in one line and then just run clang-format. :)